hashes/src/sha256t.rs

```diff
@@ -135,13 +136,33 @@ impl<T: Tag> core::hash::Hash for Hash<T> {
 
 crate::internal_macros::hash_trait_impls!(256, false, T: Tag);
 
-fn from_engine<T>(e: sha256::HashEngine) -> Hash<T>
+fn from_engine<T>(e: HashEngine<T>) -> Hash<T>
 where
     T: Tag,
 {
-    Hash::from_byte_array(sha256::Hash::from_engine(e).to_byte_array())
+    Hash::from_byte_array(sha256::Hash::from_engine(e.0).to_byte_array())
 }
 
+/// Engine to compute SHA256t hash function.
+#[derive(Clone)]
```

Comment:
`Debug` also?

Reply:
All the hash engines are missing `Debug`. I noticed this yesterday working on #3927 and wondered if there was some reason. Nice coincidence. #4015